### ISSUE TITLE
Disk write activity detected when executing a number of SELECT queries for database with Forced Write = ON

### ISSUE THREAD

**Issue by ccy** [FirebirdSQL/firebird#7364]
I am using Firebird 3.0.10 x64 in Windows 11.  The database has Forced Write = ON.

I notice when I execute a number of SELECT queries for 20 seconds:
```
start transaction (read committed)
SELECT Name from Table where Key=%d
commit transaction
```

I can notice the disk page in task manager showing a ~20 seconds disk write activities.

If I set FORCED WRITE to off and perform same task.  The disk write activities are completely silent.

Isn't the SELECT queries work as read only and shall not incur disk write operations?

**Comment by aafemt** [comment]
No. Garbage collection and other unavoidable changes in database are performed. It is not a bug.

**Comment by ccy** [comment]
In the ~20s duration, there are ~3MB/s contantly write operations.

**Comment by asfernandes** [comment]
What about transactions? How many are you starting?

**Comment by ccy** [comment]
About to 5,000 read committed transactions

**Comment by hvlad** [comment]
Every transaction forces write of header page to disk at the own start.
Every transaction that written some data, forces write of TIP page to disk at the finish (commit or rollback).
The exclusion is read-only transactions. 
Start transaction as read-only and it will not force header page.
Make no changes in transaction (even in non-read-only) and it will not force TIP page to disk.
This is a bit simplified description.

**Comment by ccy** [comment]
@hvlad : Thanks for your advice.
I use read only transaction and the performance has improved by near 40% now but this is restricted to Firebird running in **Super** mode.

The disk write operations still happen for **SuperClassic** and **Classic** mode even if I use Read Only transaction.
